common/src/main/scala/org/apache/comet/CometConf.scala

```diff
@@ -376,12 +376,15 @@ object CometConf {
     .booleanConf
     .createWithDefault(false)
 
-  val COMET_CAST_STRING_TO_TIMESTAMP: ConfigEntry[Boolean] = conf(
-    "spark.comet.cast.stringToTimestamp")
-    .doc(
-      "Comet is not currently fully compatible with Spark when casting from String to Timestamp.")
-    .booleanConf
-    .createWithDefault(false)
+  val COMET_CAST_ALLOW_INCOMPATIBLE: ConfigEntry[Boolean] =
+    conf("spark.comet.cast.allowIncompatible")
+      .doc(
+        "Comet is not currently fully compatible with Spark for all cast operations. " +
+          "Set this config to true to allow them anyway. See compatibility guide " +
+          "for more information.")
+      .booleanConf
+      // TODO change this to false and set this config explicitly in tests where needed
+      .createWithDefault(true)
```

Comment:
I will create a separate PR to enable this and update any tests that need it, after this PR is merged